core/trino-main/src/main/java/io/trino/execution/TableInfo.java

```diff
@@ -86,13 +86,14 @@ public static Map<PlanNodeId, TableInfo> extract(Session session, Metadata metad
 
     private static TableInfo extract(Session session, Metadata metadata, TableScanNode node)
     {
-        TableSchema tableSchema = metadata.getTableSchema(session, node.getTable());
```

Comment:
Do we still nee `getTableSchema`? It was supposed to be faster than getting table metadata and it is still slow anyway?

Reply:
good question. `io.trino.metadata.Metadata#getTableSchema` is still used & not going away

yes, `io.trino.metadata.Metadata#getTableSchema` is faster than `getTableMetadata`, but `getTableName` is even faster (usually no-op, since table handles typically contain the name)